.gitignore

```diff
@@ -1,4 +1,8 @@
+# terraform
 **/.terraform/*
 *.tfstate
 *.tfstate.*
 crash.log
+
+# pycharm
+.idea/**
```

Comment:
I haven't seen this syntax used this way before.   I believe `.idea/` works the same way.   

From the docs (https://git-scm.com/docs/gitignore#_pattern_format):
> Two consecutive asterisks ("**") in patterns matched against full pathname may have special meaning:
>
> A leading "**" followed by a slash means match in all directories. For example, "**/foo" matches file or directory "foo" anywhere, the same as pattern "foo". "**/foo/bar" matches file or directory "bar" anywhere that is directly under directory "foo".
>
> A trailing "/**" matches everything inside. For example, "abc/**" matches all files inside directory "abc", relative to the location of the .gitignore file, with infinite depth.
>
> A slash followed by two consecutive asterisks then a slash matches zero or more directories. For example, "a/**/b" matches "a/b", "a/x/b", "a/x/y/b" and so on.
>
> Other consecutive asterisks are considered regular asterisks and will match according to the previous rules.


`**/.terraform/*` makes sense because it matches a `.terraform` dir at any depth.  Although are they located at multiple depths?

Reply:
Ok, I was just freewheeling. The terraform syntax I got from gitignore.io so I'm trusting it.